plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java

```diff
@@ -667,7 +667,7 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
                         RedshiftClient::writeChar));
 
             case Types.VARCHAR: {
-                int length = type.columnSize().orElse(REDSHIFT_MAX_VARCHAR);
+                int length = type.columnSize().orElse(VarcharType.MAX_LENGTH);
```

Comment:
this change is required as synthetic column has unbounded length.

It was brought up in https://github.com/trinodb/trino/pull/23808#discussion_r1824641730
not sure if we should continue allowing Trino query which may fail in Redshift.

Reply:
@Praveen2112 can you think of any potential query that I could try which might work in Trino but not in Redshift?